negy-gateway/src/main.rs

```diff
@@ -22,6 +22,8 @@ struct Args {
     node_pool_endpoint: String,
     #[clap(short, long, value_parser, default_value = "3")]
     hops: usize,
+    #[clap(short, long, value_parser, default_value = "0.1.5")]
+    min_version: String,
```

Comment:
could you make this arg option? i mean `min_version: Option<String>`

Reply:
I did that (on my local copy) and that outputted this error.
> can't compare `std::string::String` with `Option<std::string::String>

Reply:
Yup the error says what it is. You can't compare String and Option<String>.

Reply:
So then I would need to convert the option back to string?

Reply:
You should do `If the min_version is None (not specified), every node is a target. If the min_version is specified, apply the filter`. It gonna be like this.

```
.filter(|n| {
    if let Some(min_version) = args.min_version {
        n.version >= min_version
    } else {
        true
    }
})
```

Reply:
Like that?